File: Stack.py
# Author: AG

class Node:
    def __init__(self,n):
        self.value = n
        self.next = None


class stack:
    def __init__ (self,n):
        self.root = Node(n)

    def push(self,n):
        node = Node(n)
        node.next = self.root
        self.root = node
        

        pass
    def pop(self):
        # n = None
        if self.isEmpty()== False:
            n = self.root
            self.root = self.root.next
        else :
            return None
        return n.value
        pass

    def printstack(self):
        n = self.root
        while(n.next!=None):
            print(n.value)
            n = n.next
        print(n.value)

    def isEmpty(self):
        count = 0
        n = self.root
        while(n!=None):
            count+=1
            n = n.next
        return count==0

# Testing part.....
new = stack(5)
new.push(6)
new.push(8)
new.push(9)
new.printstack()
print(new.pop())

File: Stack.java
//Author: AG

class Node{
    private int key;
    private Node next = null;

    public Node(int key) {
        this.key = key;
    }

    public Node getNext() {
        return next;
    }

    public int getKey() {
        return key;
    }

    public void setNext(Node next) {
        this.next = next;
    }
}

public class Stack {
    Node root;

    public Stack(int root) {
        this.root = new Node(root);
    }

    public void push(int key){
        Node n = new Node(key);
        n.setNext(root);
        this.root = n;
    }

    public int pop(){
        if(this.isEmpty()==false){
            Node n = this.root;
            this.root = this.root.getNext();
            return n.getKey();
        }
        return 0;
    }

    public boolean isEmpty(){
        return this.root == null;
    }

    public void printStack(){
        if(this.isEmpty()){
            System.out.println("null");
        }
        else{
            Node n = this.root;
            while(n!=null){
                System.out.println(n.getKey());
                n = n.getNext();
            }
        }
    }

    public static void main(String[] args) {
        // Static Stack creation
        Stack stack = new Stack(5);
        stack.push(10);
        stack.push(1);
        stack.push(11);
        stack.push(105);
        stack.push(18);
        stack.printStack();
        System.out.println(stack.pop());

    }
}
